tracer/src/Datadog.Trace/SpanContext.cs

```diff
@@ -43,7 +43,7 @@ public SpanContext(ulong? traceId, ulong spanId, SamplingPriority? samplingPrior
             : this(traceId, serviceName)
         {
             SpanId = spanId;
-            SamplingPriority = samplingPriority;
+            SamplingPriority = (int?)samplingPriority;
```

Comment:
Do we want an internal `int` based constructor? Maybe unnecessary.

Reply:
I don't think we need it. We only use this public constructor with `SamplingPriority` in tests.